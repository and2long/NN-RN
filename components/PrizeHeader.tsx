import { LinearGradient } from "expo-linear-gradient";
import { Dimensions, Image, StyleSheet, Text, View } from "react-native";
import { primaryColor } from "../constants/Colors";

export default function PrizeHeader() {
  return (
    <View>
      <View style={styles.container}>
        <Image source={require("../assets/images/prize_bg.png")} style={styles.headerBg} />
        <Text style={styles.headerTitie}>做任务 享免费排队加速</Text>
        <LinearGradient start={{ x: 0, y: 0 }} end={{ x: 1, y: 0 }} colors={["rgba(255, 255, 255, 0.3)", "rgba(255, 255, 255, 0)"]} style={styles.headerSubtitleBg}>
          <Text style={styles.headerSubTitle}>今日还可以完成50个任务</Text>
        </LinearGradient>
        <PointView />
      </View>
      <View style={styles.label}>
        <Text style={styles.labelTite}>每日任务</Text>
        <Text style={styles.labelSubTite}>任务于每日00:00点更新</Text>
      </View>
    </View>
  );
}

function PointView() {
  return (
    <View style={styles.pointView}>
      <View>
        <Text style={styles.pointTitle}>0</Text>
        <Text style={styles.pointSubTitle}>已获得免排队积分</Text>
      </View>
      <Image source={require("../assets/images/ic_nav.png")} style={styles.pointNav} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    height: Dimensions.get("window").width / 1.5 + 48
  },
  headerBg: {
    width: Dimensions.get("window").width,
    height: Dimensions.get("window").width / 1.5,
    resizeMode: "contain"
  },
  headerTitie: {
    position: "absolute",
    fontSize: 26,
    color: "white",
    marginTop: 80,
    marginLeft: 15
  },
  headerSubTitle: {
    fontSize: 14,
    color: "white"
  },
  headerSubtitleBg: {
    position: "absolute",
    height: 30,
    marginTop: 137,
    marginLeft: 15,
    paddingHorizontal: 15,
    paddingVertical: 7,
    borderTopLeftRadius: 15,
    borderBottomLeftRadius: 15
  },
  pointView: {
    backgroundColor: "white",
    marginHorizontal: 15,
    paddingHorizontal: 20,
    borderRadius: 12,
    height: 96,
    width: Dimensions.get("window").width - 30,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    position: "absolute",
    marginTop: Dimensions.get("window").width / 1.5 - 48
  },
  pointTitle: {
    color: primaryColor,
    fontSize: 32,
    fontWeight: "bold",
    fontStyle: "italic"
  },
  pointSubTitle: {
    color: primaryColor,
    fontSize: 16,
    fontWeight: "bold",
    marginTop: 5
  },
  pointNav: {
    width: 66,
    height: 66,
  },
  label: {
    height: 50,
    flexDirection: "row",
    justifyContent: "space-between",
    paddingHorizontal: 15,
    alignItems: "center",
  },
  labelTite: {
    fontSize: 16,
    fontWeight: "bold"
  },
  labelSubTite: {
    fontSize: 14,
    color: "#B2B2B2"
  }
})