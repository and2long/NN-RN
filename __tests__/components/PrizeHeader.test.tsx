import renderer from 'react-test-renderer';
import PrizeHeader from '../../src/components/PrizeHeader';

it(`renders correctly`, () => {
  const tree = renderer.create(<PrizeHeader taskCount={0} point={0} />).toJSON();
  expect(tree).toMatchSnapshot();
})