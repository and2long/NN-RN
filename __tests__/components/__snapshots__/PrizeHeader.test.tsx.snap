// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#eee",
        },
        Object {
          "height": 548,
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 500,
          "resizeMode": "contain",
          "width": 750,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "color": "white",
            "fontSize": 26,
            "marginLeft": 15,
            "marginTop": 80,
            "position": "absolute",
          },
        ]
      }
    >
      做任务 享免费排队加速
    </Text>
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        Object {
          "colors": Array [
            1308622847,
            16777215,
          ],
          "endPoint": Array [
            1,
            0,
          ],
          "locations": undefined,
          "startPoint": Array [
            0,
            0,
          ],
        }
      }
      style={
        Object {
          "borderBottomLeftRadius": 15,
          "borderTopLeftRadius": 15,
          "height": 30,
          "marginLeft": 15,
          "marginTop": 137,
          "paddingHorizontal": 15,
          "paddingVertical": 7,
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Object {
              "color": "white",
              "fontSize": 14,
            },
          ]
        }
      >
        今日还可以完成0个任务
      </Text>
    </ViewManagerAdapter_ExpoLinearGradient>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "alignItems": "center",
            "borderRadius": 12,
            "flexDirection": "row",
            "height": 96,
            "justifyContent": "space-between",
            "marginHorizontal": 15,
            "marginTop": 452,
            "paddingHorizontal": 20,
            "position": "absolute",
            "width": 720,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "backgroundColor": "#0000",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "#FD7C4B",
                "fontSize": 32,
                "fontStyle": "italic",
                "fontWeight": "bold",
              },
            ]
          }
        >
          0
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "#FD7C4B",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginTop": 5,
              },
            ]
          }
        >
          已获得免排队积分
        </Text>
      </View>
      <Image
        source={1}
        style={
          Object {
            "height": 66,
            "width": 66,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#eee",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-between",
          "paddingHorizontal": 15,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          },
        ]
      }
    >
      每日任务
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "color": "#B2B2B2",
            "fontSize": 14,
          },
        ]
      }
    >
      任务于每日00:00点更新
    </Text>
  </View>
</View>
`;
